WordPress/Classes/ViewRelated/Gutenberg/GutenbergViewController.swift

```diff
@@ -385,6 +387,8 @@ class GutenbergViewController: UIViewController, PostEditor, FeaturedImageDelega
 
     override func viewWillDisappear(_ animated: Bool) {
         super.viewWillDisappear(animated)
+        // Resets the action sheet's tintColor to `.primary` to prevent `.editorPrimary` being set as the tintColor for UI elements outside of the editor.
+        UIView.appearance(whenContainedInInstancesOf: [UIAlertController.self]).tintColor = .primary
```

Comment:
The need to mutate and reset this change is unfortunate and obscure. I am curious if more experience iOS developers have idea for an alternative approach.

Reply:
I wonder if we can avoid this by targeting `AztecNavigationController`. [[more context](https://github.com/wordpress-mobile/WordPress-iOS/pull/19113/files#r937295053)]

Reply:
@twstokes, thank you for taking a look at this! 

I tried `UIView.appearance(whenContainedInInstancesOf: [UIAlertController.self, AztecNavigationController.self]).tintColor = .primary`, but it doesn't work locally for me. 

I checked the hierarchy and I believe that `AztecNavigationController` is a 'sibling' view to `UIAlertController`, rather than a parent view:

<img width="815" alt="Screenshot 2022-08-04 at 09 48 39" src="https://user-images.githubusercontent.com/2998162/182805655-6fa6be6e-b9bc-4b67-b565-763702f962b5.png">

Is there perhaps another way to target the `UIAlertController` that I'm missing?

Reply:
> I checked the hierarchy and I believe that AztecNavigationController is a 'sibling' view to UIAlertController, rather than a parent view

Good catch! When I did my smoke test this code was still active so I thought the appearance proxy umbrella covered it. 👍 

> Is there perhaps another way to target the UIAlertController that I'm missing?

I'll look into this.

Reply:
👋 @SiobhyB - This doesn't solve all the UIAlertController cases, but what do you think about adding `alert.view.tintColor = .editorPrimary` here to explicitly set the "More Sheet" tint color?

https://github.com/wordpress-mobile/WordPress-iOS/blob/9f4c72a6d8d374b643b20556f0101dfb50e6a035/WordPress/Classes/ViewRelated/Gutenberg/GutenbergViewController%2BMoreActions.swift#L16-L18

Reply:
For `UIAlertController`s created on the RN side (e.g. Image block -> Add Image -> Choose image), is there a reason why we're not trying to enforce the tint from there (if that's possible)? Or perhaps RN has no concept of the branding colors?

Reply:
If this can't be controlled / set from the RN side, I currently can't think of a better solution than what you have @SiobhyB. It's fascinating to me that the bottom sheet being presented by the editor isn't a descendant.

Reply:
> For UIAlertControllers created on the RN side (e.g. Image block -> Add Image -> Choose image), is there a reason why we're not trying to enforce the tint from there (if that's possible)? Or perhaps RN has no concept of the branding colors?

Woah, yes, I just tested this quickly and it seems possible! It seems such an obvious solution, not sure why I went down such a complicated road with it now. 🤦‍♀️ Thanks so much @twstokes, I'll work on a Gutenberg PR.

Reply:
I've gone ahead to create that PR at https://github.com/WordPress/gutenberg/pull/42996 and have removed the (now redundant) calls within iOS in https://github.com/wordpress-mobile/WordPress-iOS/pull/19113/commits/a67b94b00015a2b56b4301550e4f73510837f19a.